Dijkstra Algorithm/README.md

```diff
@@ -0,0 +1,287 @@
+
+The table below represents graph state:
+
+|                           |  A  |  B  |  C  |  D  |  E  |
+|:------------------------- |:---:|:---:|:---:|:---:|:---:|
+| Visited                   |  F  |  F  |  F  |  F  |  F  |
+| Path Length From Start    | inf | inf | inf | inf | inf |
+| Path Vertices From Start  | [ ] | [ ] | [ ] | [ ] | [ ] |
+
+>inf is equal infinity, which basically means, that algorithm doesn't know how far away is this vertex from start one.
+
+>F states for False
+
+>T states for True
+
+To initialize our graph we have to set source vertex path length from source vertex to 0, and append itself to path vertices from start.
+
+|                           |  A  |  B  |  C  |  D  |  E  |
+|:------------------------- |:---:|:---:|:---:|:---:|:---:|
+| Visited                   |  F  |  F  |  F  |  F  |  F  |
+| Path Length From Start    |  0  | inf | inf | inf | inf |
+| Path Vertices From Start  | [A] | [ ] | [ ] | [ ] | [ ] |
+
+Great, now our graph is initialized and we can pass it to the Dijkstra's algorithm.
+
+But before we will go through all process side by side read this explanation.
+The algorithm repeats following cycle until all vertices are marked as visited.
+Cycle:
+1. From the non-visited vertices the algorithm picks a vertex with the shortest path length from the start (if there are more than one vertex with the same shortest path value, then algorithm picks any of them)
+2. The algorithm marks picked vertex as visited.
+3. The algorithm check all of its neighbors. If the current vertex path length from the start plus an edge weight to a neighbor less than the neighbor current path length from the start, than it assigns new path length from the start to the neihgbor.
+When all vertices are marked as visited, the algorithm's job is done. Now, you can see the shortest path from the start for every vertex by pressing the one you are interested in.
+
+Okay, let's start!
+Let's follow the algorithm's cycle and pick the first vertex, which neighbors we want to check.
+All our vertices are not visited, but there is only one has the smallest path length from start - A. This vertex is the first one, which neighbors we will check.
+First of all, set this vertex as visited 
+
+A.visited = true
+
+<img src="Images/image2.png" height="250" />
+
+After this step graph has this state:
+
+|                           |  A  |  B  |  C  |  D  |  E  |
+|:------------------------- |:---:|:---:|:---:|:---:|:---:|
+| Visited                   |  T  |  F  |  F  |  F  |  F  |
+| Path Length From Start    |  0  | inf | inf | inf | inf |
+| Path Vertices From Start  | [A] | [ ] | [ ] | [ ] | [ ] |
+
+## Step 1
+
+Then we check all of its neighbors. 
+If checking vertex path length from start + edge weigth is smaller than neighbor's path length from start, then we set neighbor's path length from start new value and append to its pathVerticesFromStart array new vertex: checkingVertex. Repeat this action for every vertex.
+
+for clarity:
```

Comment:
Diagrams here will help a lot. You could do something like this:

<img width="651" alt="screen shot 2017-05-01 at 5 31 24 pm" src="https://cloud.githubusercontent.com/assets/11384608/25599806/085b0b54-2e94-11e7-9ef1-f0f48e951d44.png">

This diagram focuses the reader on what you're currently trying to show (the path from A to B), while phasing out everything else. You'll leave the goal (Vertex E) in plain sight to remind the reader of the goal.